### DISCUSSION TITLE
Adding un-ack functionality to ObservableQueue

### DISCUSSION THREAD

**Discussion by YoBeMo** [Netflix/conductor-community#50] [Ideas]
The default event process doesn't act on unhandled exceptions.

That means that in some use cases messages are de-queued, consumed by _Conductor_ and fail somewhere in the process.
In such cases, no action is made against the queue message itself, meaning that a message can remain _un-acked_.

In _AMQP_, messages that are extracted from a queue and are not _ack\nacked_ will result in channel disconnection after some interval (the default is 30 min).
We thought about adding a _nack_ functionality, so in such cases, a message will be _nacked_ and would not remain hanging without any proper handling.

What do you think about this approach?

**Comment by v1r3n** [comment]
Hi @YoBeMo this is a good approach, feel free to a send a PR.
Relevant issue:
https://github.com/Netflix/conductor-community/issues/44
